src/js/services/integrations-service.js

```diff
@@ -37,17 +38,57 @@ export function resetTwilioAttributes() {
 export function fetchTwilioAttributes() {
     const {user: {clients, pendingClients}} = store.getState();
     const client = clients.find((client) => client.platform === 'twilio');
-    const pendingClient = pendingClients.find((client) => client.platform === 'twilio');
+    const pendingClient = pendingClients && pendingClients.find((client) => client.platform === 'twilio');
 
     if (client) {
         updateTwilioAttributes({
             linkState: 'linked',
-            number: client.displayName
+            appUserNumber: client.displayName
         });
     } else if (pendingClient) {
         updateTwilioAttributes({
             linkState: 'pending',
-            number: pendingClient.displayName
+            appUserNumber: pendingClient.displayName
         });
     }
 }
+
+export function linkTwilioChannel(userId, data) {
+    return core().appUsers.link.linkChannel(userId, data)
+        .then((appUser) => {
+            return immediateUpdate(appUser);
```

Comment:
why are we updating the user at all? if the server is returning the latest version of the appUser we should just be dispatching it to the store and that's it? There's nothing to be updated, right?